mbuild/tests/test_compound.py

```diff
@@ -1332,6 +1332,157 @@ def test_none_charge(self):
     def test_energy_minimize(self, octane):
         octane.energy_minimize()
 
+    @pytest.mark.skipif(
+        not has_openbabel, reason="Open Babel package not installed"
+    )
+    def test_energy_minimize_shift_com(self, octane):
+        com_old = octane.pos
+        octane.energy_minimize()
+        # check to see if COM of energy minimized Compound
+        # has been shifted back to the original COM
+        assert np.allclose(com_old, octane.pos)
+
+    @pytest.mark.skipif(
+        not has_openbabel, reason="Open Babel package not installed"
+    )
+    def test_energy_minimize_shift_anchor(self, octane):
+        anchor_compound = octane.labels["chain"].labels["CH3"][0]
+        pos_old = anchor_compound.pos
+        octane.energy_minimize(anchor=anchor_compound)
+        # check to see if COM of the anchor Compound
+        # has been shifted back to the original COM
+        assert np.allclose(pos_old, anchor_compound.pos)
+
+    @pytest.mark.skipif(
+        not has_openbabel, reason="Open Babel package not installed"
+    )
+    def test_energy_minimize_fix_compounds(self, octane):
+        methyl_end0 = octane.labels["chain"].labels["CH3"][0]
+        methyl_end1 = octane.labels["chain"].labels["CH3"][1]
+        carbon_end = octane.labels["chain"].labels["CH3"][0].labels["C"][0]
+        not_in_compound = mb.Compound(name="H")
+
+        # fix the whole molecule and make sure positions are close
+        # given stochastic nature and use of restraining springs
+        # we need to have a pretty loose tolerance for checking
+        old_com = octane.pos
+        octane.energy_minimize(
+            fixed_compounds=octane, shift_com=False, constraint_factor=1e6
+        )
+        assert np.allclose(octane.pos, old_com, rtol=1e-2, atol=1e-2)
+
+        # primarily focus on checking inputs are parsed correctly
+        octane.energy_minimize(fixed_compounds=[octane])
+        octane.energy_minimize(fixed_compounds=carbon_end)
+        octane.energy_minimize(fixed_compounds=methyl_end0)
+        octane.energy_minimize(fixed_compounds=[methyl_end0])
+        octane.energy_minimize(
+            fixed_compounds=[methyl_end0, (True, True, True)]
+        )
+        octane.energy_minimize(
+            fixed_compounds=[methyl_end0, (True, True, False)]
+        )
+        octane.energy_minimize(
+            fixed_compounds=[methyl_end0, (True, False, False)]
+        )
+        octane.energy_minimize(
+            fixed_compounds=[methyl_end0, (False, False, False)]
+        )
+
+        octane.energy_minimize(fixed_compounds=[methyl_end0, methyl_end1])
+        octane.energy_minimize(fixed_compounds=[[methyl_end0], [methyl_end1]])
+        octane.energy_minimize(
+            fixed_compounds=[
+                [methyl_end0, (True, True, True)],
+                [methyl_end1, (True, True, True)],
+            ]
+        )
+
+        with pytest.raises(MBuildError):
```

Comment:
Most of the missing lines of coverage are just error screening. You could potentially add a bunch more of these style checks to bump up your code coverage and I think this would hit the requirements.

Reply:
I added in a bunch more to check various ways to cause errors.